page-tracking/src/page_tracker.rs

```diff
@@ -199,7 +199,17 @@ impl PageTracker {
     {
         let mut page_tracker = self.inner.lock();
         let info = page_tracker.get_mut(page.addr()).unwrap();
-        info.assign(owner, PageState::Mapped)?;
+        let was_locked = info.is_locked();
+        if !was_locked {
+            info.lock_for_assignment()?;
+        }
```

Comment:
Why is the page being locked here if it wasn't previously? Wouldn't this silently hide a bug where we attempted to assign an unlocked converted page?

Reply:
I should have added an comment that It was a crutch in the draft PR for debugging purposes.